clang/include/clang/StaticAnalyzer/Core/PathSensitive/SMTConv.h

```diff
@@ -570,23 +572,35 @@ class SMTConv {
   // TODO: Refactor to put elsewhere
   static inline QualType getAPSIntType(ASTContext &Ctx,
                                        const llvm::APSInt &Int) {
-    return Ctx.getIntTypeForBitwidth(Int.getBitWidth(), Int.isSigned());
+    QualType Ty;
+    if (!(Ty = Ctx.getIntTypeForBitwidth(Int.getBitWidth(), Int.isSigned()))
+             .isNull())
```

Comment:
```suggestion
    QualType Ty = Ctx.getIntTypeForBitwidth(Int.getBitWidth(), Int.isSigned());
    if (!Ty.isNull())
```

Reply:
Why not using the init-if as I suggested?
```suggestion
    if (QualType Ty = Ctx.getIntTypeForBitwidth(Int.getBitWidth(), Int.isSigned()); !Ty.isNull())
```

Reply:
@steakhal , because I get this error if I follow the precise suggestion. So I modified a little bit to meet what I thought you wanted.

```
In file included from <base>/clang/lib/StaticAnalyzer/Core/Z3CrosscheckVisitor.cpp:18:
<base>/clang/include/clang/StaticAnalyzer/Core/PathSensitive/SMTConv.h: In static member function ‘static clang::QualType clang::ento::SMTConv::getAPSIntType(clang::ASTContext&, const llvm::APSInt&)’:
<base>/clang/include/clang/StaticAnalyzer/Core/PathSensitive/SMTConv.h:585:5: error: ‘Ty’ was not declared in this scope
  585 |     Ty = Ctx.getIntTypeForBitwidth(Pow2DestWidth, Int.isSigned());
      |     ^~
ninja: build stopped: subcommand failed.
```

Reply:
Alright then. Then at least move the initializer out of the if condition, because objectively assignments within conditions are uncommon. And if we are there we could make `Ty` const to at least imply that we won't mutate this afterwards.